### ISSUE TITLE
SlideMenu: Issue with scrollbar

### ISSUE THREAD

**Issue by wolnynick** [primefaces/primereact#3398]
### Describe the bug

When we navigate for example from root into nested menu which has more items and where scrollbar is required and back to root then scrollbar is still visible.

![image](https://user-images.githubusercontent.com/40297940/193402339-624afcb2-cb2c-44bc-92c2-69d2cf5ee851.png)

![image](https://user-images.githubusercontent.com/40297940/193402358-8ce3aeb1-f824-49cf-a4b9-510ccf908117.png)

![image](https://user-images.githubusercontent.com/40297940/193402380-1d2c54dd-5047-4241-8ae5-ed8ad1f3f7a5.png)

![image](https://user-images.githubusercontent.com/40297940/193402424-5bf71d25-90f8-4b17-8283-1c9a0950c24f.png)

Maybe last open subMenu shouldn't be rendered? (I know that this solution is only for one level nested menus)

```js
                                {!levelState && <SlideMenuSub menuProps={props} model={props.model} root index={0} menuWidth={props.menuWidth} effectDuration={props.effectDuration} level={levelState} parentActive={levelState === 0} onForward={navigateForward} /> }
```

### PrimeReact version

8.6.1

### React version

18.x

### Language

TypeScript

### Build / Runtime

Vite

### Browser(s)

Chrome

**Comment by melloware** [comment]
Can you put together a Code Sandbox so I can see the problem? https://codesandbox.io/s/primereact-test-forked-bbns8k

**Comment by wolnynick** [comment]
@melloware 
Here's the [sandbox](https://codesandbox.io/s/primereact-test-forked-bmi06m?file=/src/index.js).

Except changes in code I made also changes in styles.css. I implemented some mine code into this, but I think this error is also in basic version.

First step - Open menu on root level (no scroll needed):

![image](https://user-images.githubusercontent.com/40297940/193449648-b1af6d59-a823-41b9-ae09-bdf55289e765.png)

Second step - After click on first root element we can go to the nested level with 21 elements (make sure your screen isn't big enough to display all). And here our scroll is needed and rendered.

![image](https://user-images.githubusercontent.com/40297940/193449702-af543fc3-2777-4583-a7e8-4307336e5334.png)

Third step - Go back into root:

![image](https://user-images.githubusercontent.com/40297940/193449729-b8274233-2bb5-4a22-a8ea-1f6da00bb9c6.png)

And there is still scroll from last submenu. (This submenu is still in DOM)


I think that can be fixed by removing <SlideMenuSub /> from DOM if we are not on its level.
